### ISSUE TITLE
[quotemark: backtick] Incorrectly flags string literals in export line

### ISSUE THREAD

**Issue by ericbf** [palantir/tslint#4588]
### Bug Report

- __TSLint version__: 5.14.0
- __TypeScript version__: Version 3.3.3333
- __Running TSLint via__: Atom
- __Related issues__: #4514 

#### TypeScript code being linted

```ts
export * from "./some-module"
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "quotemark": "backtick"
    }
}
```

#### Actual behavior

The quotes get flagged (and changed to backticks if fix is enabled).

#### Expected behavior

The quotes should remain as double/single quotes, as typescript/ecma doesn't allow backticks here.

**Comment by ericbf** [comment]
Here's another example:

**Code being linted**
```typescript
const array: SomeType<string, "hello"> = []
const obj: {
	"kebab-case": string
} = {
	"kebab-case": "value"
}
```

**Expected result of fix**
```typescript
const array: SomeType<string, "hello"> = []
const obj: {
	"kebab-case": string
} = {
	"kebab-case": `value`
//                    ~~~~~~~ [" should be `]
}
```

**Actual result of fix**
```typescript
const array: SomeType<string, `hello`> = []
//                            ~~~~~~~ [" should be `]
const obj: {
	`kebab-case`: string
//      ~~~~~~~~~~~~ [" should be `]
} = {
	"kebab-case": "value"
//                    ~~~~~~~ This one is ignored, but should be flagged
}
```

**Problem summary**

1. The fixing/flagging here flags some things that when changed breaks typescript syntax and doesn't flag other instances where it should be flagged.
1. Typescript required that the string literal type passed to the generic type be single/double, but the implementation of the backtick rule does not make that distinction. Seems that node doesn't have a clear-cut type though to exclude it.
1. For the type definition, the kebab case property name has to remain single/double quotes, or typescript breaks as well. What is the associated node type there? Mucking in the code trying to PR for this, seems that the type definition property name here doesn't have a clean-cut node type either.
1. The string literal in the property value assignment is not flagged, but it should. Seems that looking at the parent node (`"kebab-case": "value"`) gives you "PropertyAssignment" for both the property key and the property value, while the key should be ignored in this case, the value should not. How would one implement this part when they don't have their own node types?

**Comment by ericbf** [comment]
Actually, I think I got something. PR soon.
